src/lib/viewers/doc/DocBaseViewer.js

```diff
@@ -153,6 +155,33 @@ class DocBaseViewer extends BaseViewer {
         super.destroy();
     }
 
+    /**
+     * Converts a value and unit to page number
+     *
+     * @return {number|undefined} a page number > 0
+     */
+    getStartPage() {
```

Comment:
I recommend calling this in pagesInitHandler where we already handle logic like this. We can also do validation at the same time, and fall back to a cached value that we may have stored.

Reply:
+1 - see in pagesinitHandler():

```
// Set current page to previously opened page or first page
        this.setPage(this.getCachedPage());
```

Like Jeremy mentioned, `startAt` should take precedence over whatever value is stored in local storage, and if there's no `startAt`, fall back to page in local storage.

Reply:
Moved the call to pagesInitHandler